From 9dc7ae7d52e9d44935fb49f980d9f1280594d8b1 Mon Sep 17 00:00:00 2001
From: Nick Morrott <knowledgejunkie@gmail.com>
Date: Fri, 1 Mar 2019 00:01:48 +0000
Subject: [PATCH 03/13] tv_imdb: fix typo in documentation

---
 filter/tv_imdb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/filter/tv_imdb b/filter/tv_imdb
index 0a9c19e7f..ee0b2d887 100755
--- a/filter/tv_imdb
+++ b/filter/tv_imdb
@@ -151,7 +151,7 @@ How and what to look up needs to be option driven.
 Needs some more controls for fine tuning "close" matches. For
 instance, currently it looks like the North America grabber only has
 date entries for movies, but the imdb.com data contains made for video
-movies as well as as real movies, ot is itE<39>s possible to get the
+movies as well as as real movies, so itE<39>s possible to get the
 wrong data to be inserted. In this case we may want to say "ignore tv
 series" and "ignore tv mini series".  Along with this, weE<39>d want
 to define what a "close" match is. For instance does a movie by the
-- 
2.21.0

