From ea01fcb293b1d95ce89fe055bef38b54253de26e Mon Sep 17 00:00:00 2001
From: Nick Morrott <knowledgejunkie@gmail.com>
Date: Tue, 8 Sep 2020 03:33:16 +0100
Subject: [PATCH 04/50] tv_imdb: fix some typos

---
 filter/tv_imdb | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/filter/tv_imdb b/filter/tv_imdb
index ee0b2d88..0addb452 100755
--- a/filter/tv_imdb
+++ b/filter/tv_imdb
@@ -32,9 +32,9 @@ XMLTV::IMDB package.
 
 B<--output FILE> write to FILE rather than standard output.
 
-B<--with-keywords> include IDMb keywords in the output file.
+B<--with-keywords> include IMDb keywords in the output file.
 
-B<--with-plot> include IDMb plot summary in the output file.
+B<--with-plot> include IMDb plot summary in the output file.
 
 B<--actors NUMBER> number of actors from IMDb to add (default=3).
 
@@ -115,7 +115,7 @@ the '--download' flag and be prompted for what you need to download by
 hand.  See <http://www.imdb.com/interfaces> for the download sites.
 Then once you have the files rerun without '--download'.
 
-Note: '--prepStage' sucks a bit of memeory, but you can run each
+Note: '--prepStage' sucks a bit of memory, but you can run each
 prepStage separately by running --prepStage with each of the stages
 (see --help for details).
 
@@ -158,8 +158,8 @@ to define what a "close" match is. For instance does a movie by the
 same title with a date out by 1 year or 2 years considered a match
 (currently weE<39>re using 2).
 
-Nice to haves include: verification/addition of programe MPAA/VCHIP ratings,
-addition of imdb.com user ratings (by votes) to programes. Potenially we
+Nice to haves include: verification/addition of programme MPAA/VCHIP ratings,
+addition of imdb.com user ratings (by votes) to programmes. Potentially we
 could expand to include "country of origin", "description", "writer" and
 "producer" credits, maybe even "commentator".
 
-- 
2.29.2

