From 9b90af5866b4e9e61ed2b9ba5fa291ed4d4e8876 Mon Sep 17 00:00:00 2001
From: Beavis <beavis@scaryflop.com>
Date: Fri, 8 Mar 2019 19:10:16 +0100
Subject: [PATCH 07/13] fix url (https) and fix graber

---
 grab/fr/tv_grab_fr | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/grab/fr/tv_grab_fr b/grab/fr/tv_grab_fr
index 027995096..0f59d6663 100755
--- a/grab/fr/tv_grab_fr
+++ b/grab/fr/tv_grab_fr
@@ -141,7 +141,7 @@ use IO::File;
 #***************************************************************************
 # Main declarations
 #***************************************************************************
-my $ROOT_URL                = 'http://www.telestar.fr';
+my $ROOT_URL                = 'https://www.telestar.fr';
 my $GRID_FOR_CHANNEL        = "$ROOT_URL/programme-tv/";
 my $GRID_BY_CHANNEL_PER_DAY = "$ROOT_URL/programme-tv/grille-chaine/";
 
@@ -916,7 +916,7 @@ sub get_channels {
         return %bouquet_channels;
     }
 
-    foreach my $cellTree ( $t->look_down( "_tag", "div", "class", "channel" ) ) {
+    foreach my $cellTree ( $t->look_down( "_tag", "td", "class", "channel" ) ) {
         my $tag = $cellTree->look_down( "_tag", "a" );
         if (defined $tag) {
             my $progurl = $tag->attr('href');
-- 
2.21.0

