From 370d3e50a034509a3ae8fcbed1df00380c673120 Mon Sep 17 00:00:00 2001
From: Nick Morrott <knowledgejunkie@gmail.com>
Date: Sat, 6 Apr 2019 03:07:36 +0100
Subject: [PATCH 10/13] README.win32: drop details of Windows 98 error with
 Perl 5.8

---
 doc/README.win32 | 26 --------------------------
 1 file changed, 26 deletions(-)

diff --git a/doc/README.win32 b/doc/README.win32
index 27484a9ce..7630fac56 100644
--- a/doc/README.win32
+++ b/doc/README.win32
@@ -18,9 +18,6 @@ for tv_grab_is, tv_grab_pt_meo and tv_grab_it_dvb, If you need one
 of those you'll need to install Perl and the necessary modules and
 use the full distribution.
 
-Some Win98 users get errors in Perl58.dll when running tv_grab_na_dd. See
-below for more info.
-
 tv_grab_se_swedb sometimes fails to work on Windows if there are spaces
 in the path to your home-directory. This can be avoided by setting
 the environment variable HOME to a path without spaces (e.g. c:\home).
@@ -91,29 +88,6 @@ To get around this, use %% to pass a % to the application. (ex. %%Y%%M )
 If you *DO* want to insert a shell variable, you can do so by surrounding
 it with percents. (ex %HOME% )
 
-* crash in with PERL58.DLL
-
-When using tv_grab_na_dd, some users are experiencing a crash in the
-Perl58.DLL.
-
-I had a similar problem when I had an older version of Perl installed
-as well as XMLTV. It seems to cause DLL confusion and eventually a program
-crash.
-
-I fixed my system when I removed the old Perl.  Re-installing Perl
-didn't cause the problem to return.
-
-Before that, I used the work-around below (I could fetch 3 days)
-
-tv_grab_na_dd --output=a.xml --days 3
-tv_grab_na_dd --output=b.xml --days 3 --offset 3
-tv_grab_na_dd --output=c.xml --days 1 --offset 6
-tv_cat a.xml b.xml c.xml >guide.xml
-
-I really thought this problem would be a fluke and/or easily solved
-when I solved it for myself.  That doesn't look to be the case.  I'm
-looking for a solution.
-
 * Proxy servers
 
 Proxy server support is provide by the LWP modules.
-- 
2.21.0

