From 8f61ae35af813ecd0e381fc4d9c199381c8d10d6 Mon Sep 17 00:00:00 2001
From: Honir <honir@c4b.co.uk>
Date: Wed, 23 Dec 2020 17:02:09 +0000
Subject: [PATCH 20/50] Unbreak parsing of keywords file

---
 lib/IMDB.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/IMDB.pm b/lib/IMDB.pm
index 1e794ff9..2c36b2b9 100644
--- a/lib/IMDB.pm
+++ b/lib/IMDB.pm
@@ -2007,7 +2007,7 @@ sub readKeywords($$$$)
 	    }
 	    last;
 	}
-	elsif ( $lineCount > 100000 ) {
+	elsif ( $lineCount > 150000 ) {		# line 101935 as at 2020-12-23
 	    $self->error("$file: stopping at line $lineCount, didn't see \"THE KEYWORDS LIST\" line");
 	    closeMaybeGunzip($file, $fh);
 	    return(-1);
-- 
2.29.2

