From c743c5a38d9717eb33f7d5361369a970f96c0499 Mon Sep 17 00:00:00 2001
From: Nick Morrott <knowledgejunkie@gmail.com>
Date: Thu, 31 Dec 2020 00:20:34 +0000
Subject: [PATCH 33/50] eu_xmltvse: refresh test.conf

---
 grab/eu_xmltvse/test.conf | 73 +++++++++++++++++++++++++++------------
 1 file changed, 51 insertions(+), 22 deletions(-)

diff --git a/grab/eu_xmltvse/test.conf b/grab/eu_xmltvse/test.conf
index 800f9130..628b06e9 100644
--- a/grab/eu_xmltvse/test.conf
+++ b/grab/eu_xmltvse/test.conf
@@ -8,29 +8,30 @@ channel!2bar.dazn.de
 channel!3.bluemovie.de
 channel!360tunebox.spi.pl
 channel!3sat.de
+channel!aachen.wdr.daserste.de
 channel!action.sky.de
 channel!actionhd.sky.de
 channel!adultchannel.co.uk
 channel!ae-tv.de
 channel!animalplanet.discovery.de
 channel!ar.france24.com
-channel=arte.de
+channel!arte.de
 channel!arthouse.spi.pl
 channel!arts.sky.de
 channel!asien.dw.de
-channel!at.viva.tv
 channel!atlantic.sky.de
 channel!atlantichd.sky.de
 channel!atlanticp1.sky.de
 channel!atv.at
 channel!atv2.at
-channel!bangutv.com
 channel!bbcworldnews.com
 channel!beate-uhse.tv
 channel!bfs.daserste.de
 channel!bibeltv.de
-channel!blizztv.de
+channel!bielefeld.wdr.daserste.de
+channel!bonn.wdr.daserste.de
 channel!br-alpha.daserste.de
+channel!br-klassik.daserste.de
 channel!brandnew.mtv.de
 channel!brazzerstveurope.com
 channel!bundesliga1.sky.de
@@ -58,14 +59,15 @@ channel!bw.swr.daserste.de
 channel!canal24h.rtve.es
 channel!cbsreality.tv
 channel!cinema.sky.de
-channel!cinemagic.disneychannel.de
 channel!cinemahd.sky.de
+channel!classic.vh1.se
 channel!classica.de
 channel!classics.kabel1.de
 channel!classicshd.kabel1.de
 channel!comedy.sky.de
-channel!comedycentral.at
+channel=comedycentral.at
 channel!comedycentral.de
+channel!crime.rtl.de
 channel!crimehd.rtl.de
 channel=daserste.de
 channel!de.eonline.com
@@ -76,8 +78,10 @@ channel!dmax.discovery.de
 channel!dmaxhd.discovery.de
 channel!docubox.spi.pl
 channel!doku.kabel1.de
+channel!dortmund.wdr.daserste.de
 channel!dr1.dr.dk
-channel!dr1hd.dr.dk
+channel!duisburg.wdr.daserste.de
+channel!dusseldorf.wdr.daserste.de
 channel!dw.de
 channel!eins.sky.de
 channel!einsextra.daserste.de
@@ -85,6 +89,9 @@ channel!einsfestival.daserste.de
 channel!einshd.sky.de
 channel!emotion.sky.de
 channel!emotions.sat1.de
+channel!erox.spi.pl
+channel!eroxxx.spi.pl
+channel!essen.wdr.daserste.de
 channel!euronews.com
 channel!europa.tve.es
 channel!europe.bloomberg.com
@@ -94,20 +101,30 @@ channel!europe.realitykings.com
 channel!extremesports.com
 channel!family.cinema.sky.de
 channel!familyhd.cinema.sky.de
-channel!familytv.de
 channel!fashionbox.spi.pl
 channel!fastandfun.spi.pl
 channel!fightbox.spi.pl
+channel!filmboxbasic.spi.pl
+channel!filmboxfamily.spi.pl
+channel!filmboxhd.spi.pl
+channel!filmboxplus.spi.pl
+channel!filmboxpremium.spi.pl
 channel!fixundfoxi.tv
+channel!fm4.orf.at
 channel!foxchannel.de
 channel!fr.france24.com
 channel!france24.com
 channel!fsf.fightsports.tv
-channel!ftv.com
+channel!fuel.tv
 channel!fun.prosieben.de
+channel!geo-television.de
+channel!ginx.tv
+channel!god.tv
 channel!gold.sat1.de
 channel!goldhd.sat1.de
+channel!gospel.tv
 channel!hd.13thstreet.de
+channel!hd.3sat.de
 channel!hd.anixehd.tv
 channel!hd.arte.de
 channel!hd.bibeltv.de
@@ -116,26 +133,31 @@ channel!hd.daserste.de
 channel!hd.deluxemusic.tv
 channel!hd.discovery.de
 channel!hd.disneychannel.de
+channel!hd.eurosport.de
 channel!hd.foxchannel.de
-channel!hd.ftv.com
 channel!hd.historytv.de
 channel!hd.kabel1.de
 channel!hd.kinowelt.tv
+channel!hd.mezzo.tv
+channel!hd.mtv.de
+channel!hd.mtv.se
 channel!hd.n-tv.de
 channel!hd.natgeo.de
 channel!hd.nick.de
 channel!hd.prosieben.de
 channel!hd.rtl.de
-channel!hd.rtl2.de
 channel!hd.sat1.de
 channel!hd.servustv.com
+channel!hd.servustv.de
 channel!hd.sixx.de
 channel!hd.spiegel-geschichte.tv
 channel!hd.syfy.de
 channel!hd.tele5.de
-channel!hd.tv2.dk
+channel!hd.viva.tv
 channel!hd.vox.de
+channel!hd.zdf.de
 channel!heimatkanal.de
+channel!hgtv.discovery.de
 channel!hits.sky.de
 channel!hitshd.sky.de
 channel!hr.daserste.de
@@ -148,7 +170,6 @@ channel!infokanal.zdf.de
 channel!int.kinopolska.pl
 channel!int.kinopolskamuzyka.pl
 channel!international.rt.com
-channel!international.skynews.com
 channel!jr.disneychannel.de
 channel!jr.nick.de
 channel!jukebox-tv.de
@@ -158,24 +179,32 @@ channel!kika.daserste.de
 channel!kikahd.daserste.de
 channel!kinowelt.tv
 channel!krimi.sky.de
+channel!living.rtl.de
 channel!maxx.prosieben.de
 channel!maxxhd.prosieben.de
 channel!mdr.daserste.de
 channel!mdrhd.daserste.de
+channel!mezzo.tv
 channel!motorvision.de
+channel!munster.wdr.daserste.de
+channel!mv.ndr.daserste.de
 channel!n-tv.de
 channel!n24doku.de
 channel!natgeo.de
 channel!ndr.daserste.de
 channel!ndrhd.daserste.de
+channel!nds.ndr.daserste.de
 channel!neo.zdf.de
 channel!neohd.zdf.de
 channel!nick.de
-channel!nickelodeon.at
+channel=nickelodeon.at
 channel!nitro.rtl.de
 channel!nitrohd.rtl.de
+channel!nord.bfs.daserste.de
 channel!nostalgie.sky.de
-channel=orf1.orf.at
+channel!oe1.orf.at
+channel!oe3.orf.at
+channel!orf1.orf.at
 channel!orf2.orf.at
 channel!orf3.orf.at
 channel!outdoorchannel.com
@@ -192,7 +221,8 @@ channel!polonia.tvp.pl
 channel!prosieben.de
 channel!protv.ro
 channel!puls4.at
-channel!rbb.daserste.de
+channel!radiob.orf.at
+channel!radiobremen.tv
 channel!rbb.rbb-online.de
 channel!rbbberl.rbb-online.de
 channel!rbbbra.rbb-online.de
@@ -207,14 +237,15 @@ channel!rtlplus.de
 channel!rts1.rts.ch
 channel!rts2.rts.ch
 channel!russia.rt.com
+channel!s-anhalt.mdr.daserste.de
+channel!sachsen.mdr.daserste.de
 channel!sat1.de
 channel!sd.anixehd.tv
 channel!select.sky.de
 channel!selecthd.sky.de
 channel!servustv.com
-channel!sf1.srf.ch
-channel!sf2.srf.ch
-channel!sfi.srf.ch
+channel!sh.ndr.daserste.de
+channel!siegen.wdr.daserste.de
 channel!sixx.de
 channel!spiegel-geschichte.tv
 channel!sport1.sky.de
@@ -232,7 +263,6 @@ channel!sportaustria.sky.de
 channel!sportdigital.tv
 channel!sporthd1.sky.de
 channel!sporthd10.sky.de
-channel!sporthd11.sky.de
 channel!sporthd2.sky.de
 channel!sporthd3.sky.de
 channel!sporthd4.sky.de
@@ -253,7 +283,6 @@ channel!tele5.de
 channel!tlc.discovery.de
 channel!travelchanneltv.eu
 channel!tv2.dk
-channel!tv5monde.org
 channel!universalchannel.de
 channel!urbanint.trace.tv
 channel!vh1.eu
@@ -263,5 +292,5 @@ channel!wdrhd.daserste.de
 channel!welt.de
 channel!wild.natgeo.de
 channel!wildhd.natgeo.de
+channel!wuppertal.wdr.daserste.de
 channel!xd.disneychannel.de
-channel!zdf.de
-- 
2.29.2

