From 077bcec488eac50741d3ac71b889cc3cd3beb394 Mon Sep 17 00:00:00 2001
From: Honir <honir@c4b.co.uk>
Date: Thu, 31 Dec 2020 16:25:09 +0000
Subject: [PATCH 35/50] Fix testsuite for the change to episode handling (#63)

---
 t/data-tv_imdb/lists/movies.list | 1 +
 1 file changed, 1 insertion(+)

diff --git a/t/data-tv_imdb/lists/movies.list b/t/data-tv_imdb/lists/movies.list
index 545bde76..89379710 100644
--- a/t/data-tv_imdb/lists/movies.list
+++ b/t/data-tv_imdb/lists/movies.list
@@ -30,6 +30,7 @@ Het Movie19 (1991)					1991
 Een Movie20 (1991)					1991
 Movie21 aeiouaecnssy (1991)				1991
 Movie22 dots (1991)					1991
+"The Show1" (2002)					1991
 "The Show1" (2002) {Episode title1 (#1.1)}		1991
 "The Show1" (2002) {Episode title2 (#1.2)}		1991
 "The Show1" (2002) {Episode title1 (#2.1)}		1991
-- 
2.29.2

