From 8ac9e1e5da7b5dbf4aceb16f337e65be76aeb4ae Mon Sep 17 00:00:00 2001
From: Philipp Matthias Hahn <pmhahn@pmhahn.de>
Date: Sun, 10 Jan 2021 16:18:22 +0100
Subject: [PATCH 38/50] eu-epgdata: Add channel IDs

---
 grab/eu_epgdata/channel_ids | 1 +
 1 file changed, 1 insertion(+)

diff --git a/grab/eu_epgdata/channel_ids b/grab/eu_epgdata/channel_ids
index efa8fe9c..ce0d65e9 100644
--- a/grab/eu_epgdata/channel_ids
+++ b/grab/eu_epgdata/channel_ids
@@ -180,3 +180,4 @@
 12046;sky1.sky.de:;Sky 1
 12102;esports1.sport1.de;eSports1 (ESPO1)
 12125;voxup.vox.de;VOXup (VOXUP)
+12052;sr.de;SR Fernsehen
-- 
2.29.2

