From d8ad8d2adbad6061610b4c77bb8d9a6f91c67214 Mon Sep 17 00:00:00 2001
From: Honir <honir@c4b.co.uk>
Date: Wed, 13 Jan 2021 16:52:59 +0000
Subject: [PATCH 39/50] whitespace changes

---
 filter/tv_imdb | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/filter/tv_imdb b/filter/tv_imdb
index b4f1eb8d..d134fcf9 100755
--- a/filter/tv_imdb
+++ b/filter/tv_imdb
@@ -206,24 +206,24 @@ my ($opt_help,
 	$opt_num_actors,
 	$opt_validate_title,
 	$opt_validate_year,
-    $opt_sample,
-   );
+	$opt_sample,
+	);
 
-GetOptions('help' 			=> \$opt_help,
+GetOptions('help'			=> \$opt_help,
 		'output=s'			=> \$opt_output,
-		'prepStage=s' 		=> \$opt_prepStage,
-		'imdbdir=s' 		=> \$opt_imdbDir,
-		'with-keywords' 	=> \$opt_with_keywords,
-		'with-plot'	 		=> \$opt_with_plot,
-		'movies-only' 		=> \$opt_movies_only,
-		'actors=s'	  		=> \$opt_num_actors,
-		'quiet' 			=> \$opt_quiet,
-		'download' 			=> \$opt_download,
+		'prepStage=s'		=> \$opt_prepStage,
+		'imdbdir=s'			=> \$opt_imdbDir,
+		'with-keywords'		=> \$opt_with_keywords,
+		'with-plot'			=> \$opt_with_plot,
+		'movies-only'		=> \$opt_movies_only,
+		'actors=s'			=> \$opt_num_actors,
+		'quiet'				=> \$opt_quiet,
+		'download'			=> \$opt_download,
 		'stats'		 		=> \$opt_stats,
 		'debug+'			=> \$opt_debug,
 		'validate-title=s'	=> \$opt_validate_title,
 		'validate-year=s'	=> \$opt_validate_year,
-	    'sample=s'  		=> \$opt_sample,
+		'sample=s'			=> \$opt_sample,
 		) or usage(0);
 
 usage(1) if $opt_help;
@@ -234,7 +234,7 @@ $opt_with_plot=0	 	if ( !defined($opt_with_plot) );
 $opt_num_actors=3		if ( !defined($opt_num_actors) );
 $opt_movies_only=0 		if ( !defined($opt_movies_only) );
 $opt_debug=0		 	if ( !defined($opt_debug) );
-$opt_sample=0        	if ( !defined($opt_sample) );
+$opt_sample=0			if ( !defined($opt_sample) );
 
 $opt_quiet=(defined($opt_quiet));
 if ( !defined($opt_stats) ) {
-- 
2.29.2

