diff -up xmltv-0.5.66/lib/XMLTV.pm.in~ xmltv-0.5.66/lib/XMLTV.pm.in
--- xmltv-0.5.66/lib/XMLTV.pm.in~	2014-10-21 08:55:31.000000000 +0200
+++ xmltv-0.5.66/lib/XMLTV.pm.in	2015-05-18 17:31:12.509157453 +0200
@@ -2123,7 +2123,11 @@ BEGIN {
 }
 
 BEGIN {
-  Date::Manip::Date_Init("TZ=UTC");
+    if (int(Date::Manip::DateManipVersion) >= 6) {
+	Date::Manip::Date_Init("SetDate=now,UTC");
+    } else {
+	Date::Manip::Date_Init("TZ=UTC");
+    }
 }
 
 # Override dataElement() to refuse writing empty or whitespace
diff -up xmltv-0.5.66/filter/tv_sort~ xmltv-0.5.66/filter/tv_sort
--- xmltv-0.5.66/filter/tv_sort~	2014-06-13 18:11:17.000000000 +0200
+++ xmltv-0.5.66/filter/tv_sort	2015-05-18 19:29:29.750377410 +0200
@@ -60,7 +60,12 @@ Ed Avis, ed@membled.com
 use strict;
 use XMLTV::Version '$Id: tv_sort,v 1.33 2014/06/13 16:11:17 bilbo_uk Exp $ ';
 use XMLTV::Date;
-use Date::Manip; Date_Init('TZ=UTC');
+use Date::Manip;
+if (int(Date::Manip::DateManipVersion) >= 6) {
+    Date::Manip::Date_Init("SetDate=now,UTC");
+} else {
+    Date::Manip::Date_Init("TZ=UTC");
+}
 use Getopt::Long;
 
 # We use Storable to do 'deep equality' of data structures; this
