diff -Naur xmltv-0.5.65.orig/filter/tv_to_text xmltv-0.5.65/filter/tv_to_text
--- xmltv-0.5.65.orig/filter/tv_to_text	2014-05-05 11:02:54.000000000 -0500
+++ xmltv-0.5.65/filter/tv_to_text	2014-05-25 08:47:34.448757227 -0500
@@ -31,7 +31,7 @@
 
 use strict;
 use warnings;
-use XMLTV::Version '$Id: tv_to_text,v 1.6 2014/05/05 16:02:54 bilbo_uk Exp $ ';
+use XMLTV::Version '$Id: tv_to_text,v 1.7 2014/05/25 13:36:32 bilbo_uk Exp $ ';
 use IO::File;
 use Date::Manip; Date_Init('TZ=UTC');
 use POSIX 'tmpnam';
@@ -82,7 +82,7 @@
     my ($start, $stop, $title, $sub_title, $channel, $desc) = @$_;
     $stop = '' if not defined $stop;
     $title .= " // $sub_title" if defined $sub_title;
-    print "$start--$stop\t$title\t$channel\t". ( $opt_withdesc && defined $desc ? $desc : '' ) . "\n";
+    print "$start--$stop\t$title\t$channel". ( $opt_withdesc && defined $desc ? "\t$desc" : '' ) . "\n";
     $wrote_prog = 1;
 }
 
