--- grab/eu_egon/tv_grab_eu_egon.PL
+++ grab/eu_egon/tv_grab_eu_egon.PL
@@ -2,6 +2,7 @@
 #
 
 use strict;
+use lib '.';
 
 use IO::File;
 my $out = shift @ARGV; die "no output file given" if not defined $out;
--- grab/it/tv_grab_it.PL
+++ grab/it/tv_grab_it.PL
@@ -13,6 +13,7 @@
 # code below is based on lib/XMLTV.pm.pl
 
 use strict;
+use lib '.';
 
 use File::Copy qw/copy/;
 
--- grab/se_swedb/tv_grab_se_swedb.PL
+++ grab/se_swedb/tv_grab_se_swedb.PL
@@ -3,6 +3,7 @@
 #
 
 use strict;
+use lib '.';
 
 use IO::File;
 my $out = shift @ARGV; die "no output file given" if not defined $out;
--- lib/XMLTV.pm.PL
+++ lib/XMLTV.pm.PL
@@ -5,6 +5,7 @@
 #
 
 use strict;
+use lib '.';
 sub print_list( $$ );
 
 my $out = shift @ARGV; die "no output file given" if not defined $out;
--- Makefile.PL
+++ Makefile.PL
@@ -5,6 +5,7 @@ use ExtUtils::MakeMaker;
 use Config;
 use File::Basename ();
 use File::Find;
+use lib'.';
 
 # Don't use ':config pass_through' because that requires Getopt::Long
 # 2.24 or later, and we don't have a clean way to require that.
