diff -up xmltv-0.5.58/grab/pt/tv_grab_pt.noutf8simple xmltv-0.5.58/grab/pt/tv_grab_pt
diff -up xmltv-0.5.58/Makefile.PL.noutf8simple xmltv-0.5.58/Makefile.PL
--- xmltv-0.5.58/Makefile.PL.noutf8simple	2010-10-14 17:08:51.000000000 +0200
+++ xmltv-0.5.58/Makefile.PL	2010-10-14 17:23:59.000000000 +0200
@@ -617,7 +617,6 @@ my @opt_components
        blurb => 'Grabber for Portugal',
        exes => [ 'grab/pt/tv_grab_pt' ],
        prereqs => { 'HTML::TreeBuilder' => 0,
-		    'Unicode::UTF8simple' => 0,
 		  }
      },
 
